### DISCUSSION TITLE
Ladeplanung Abfahrt über HomeAssistant setzen

### DISCUSSION THREAD

**Discussion by ZaphodB72** [marq24/ha-evcc#29] [Ideas]
Hallo,
habe EVCC als Gerät in HomeAssistant eingefügt und eine Automatisierung geschrieben, um je nach Schicht Abfahrtszeit und -SOC zu setzen. Es wird also mein Kalender ausgelesen und je nach Eintrag sollen vor allem die Zeit unterschiedlich gesetzt werden, der SOC nur zur Sicherheit bei beiden Varianten auf den gleichen Wert.
Das ganze scheitert aber daran, dass beides nur als "sensor."-Entität vorhanden ist und nicht wie bei der Ankunftsplanung als "select.", wo ich die Werte über HA ändern kann.
Kann das Team das ändern oder kann sogar ich das in HA ändern und die Werte über HA einstellbar machen?

**Comment by marq24** [comment]
Meinem Verständnis nach möchtest Du ein Ladeplan erstellen/ändern - dies funktioniert über einen Service der Integration... 

-> Adjust/create Vehicle & Loadpoint charging plan via HA-Services http://[your-ha-instance]:8123/developer-tools/service

**Reply by Steph6431** [reply]
Hallo,
auch ich würde gerne täglich um Mitternacht die Ladeplanung bei Abfahrt (konkret: das Ladeziel bei Abfahrt) je nach erwartetem PV-Forecast abändern. 
Wenn genug PV-Leistung erwartet wird soll nur wenige Strom zur günstigen Zeit geladen werden, wenn ohnehin keine PV-Leistung zu erwarten ist kann bei günstigen Tarifen auch gleich voll geladen werden.
Über den Link "http://[your-ha-instance]:8123/developer-tools/service" bin ich hier gelandet "evcc☀️🚘- Solar Charging: Abfahrt Ladeplanung für einen am Ladepunkt verbundens Fahrzeug erstellen [SOC (%)]", kenne mich aber mit den Entwicklerwerkzeugen noch nicht aus und frage mich wie ich hier ein berechnetes SOC-Ladeziel verwenden kann.
Kann mir vielleicht jemand weiterhelfen?
(Normalerweise würde ich das über eine Automation und dann über einen wenn/dann Baustein lösen...

**Reply by marq24** [reply]
... Du baust Deine HA-'Wenn-Dann'-Automatisierung und als Aktion startest Du eben den entsprechenden Service der von der evcc-ha-integraton bereitgestellt wird `Abfahrt Ladeplanung für einen am Ladepunkt verbundens Fahrzeug erstellen [SOC (%)` (und natürlich musst Du dem dann eben die richtigen Werte (data) mitgeben)...

**Reply by MrtnZ0791** [reply]
Hallo, kann mich bitte jemand darüber aufklären, wie ich dynamische Werte übergeben kann? Die Aktion `Abfahrt Ladeplanung für einen am Ladepunkt verbundenes Fahrzeug erstellen [SOC %]` erlaubt keine Templates. In den Entwicklerwerkzeugen/Aktionen scheitert der Versuch im Yaml-Modus, wenn ich hinter soc ein Templates einfüge (z.B.
`action: evcc_intg.set_vehicle_plan
  data:
    loadpoint: 1
    soc: >-
    {{ states('input_number.my_load')|float }}`
    startdate: "2025-07-20 14:33:00")
mit dem Fehler: `error: No date provided (or false data)`, wohingegen ein fester Wert, z.B.
`soc: 80`
fehlerfrei läuft. Für SOC kann ich meinetwegen mehrere Scripte mit festen Werten hinterlegen und jeweils das passende auswählen, aber wie soll ich diese Aktion aus Automationen heraus nutzen können, wenn auch das Startdatum (okay, ist im Beispiel nicht beschrieben. Das habe ich der Einfachheit halber auf SOC beschränkt) fest eingetragen werden muss?

**Reply by marq24** [reply]
@MrtnZ0791 ich denke hier liegt ein grundsätzliches Missverständnis über die Funktionalität von Diensten/Aktionen vor. Die in dem `data` Bereich angegebenen Daten müssen tatsächlich Daten sein und dürfen keine Funktionen enthalten. 

Du musst Deine Dynamik in einer Automatisierung realisieren - dort solltest Du Deine Werte 'errechnen' und dann die Aktion 'Abfahrt Ladeplanung für einen am Ladepunkt verbundenes Fahrzeug erstellen....' mit den finalen Werten aufrufen...

**Reply by MrtnZ0791** [reply]
Danke @marq24 für die sehr schnelle Antwort, doch lt. Home Assistant Dokumentation sollten Templates grundsätzlich kein Problem sein (https://www.home-assistant.io/docs/scripts/perform-actions#use-templates-to-determine-the-attributes).
Doch auch wenn es hier nicht umgesetzt ist verstehe ich nicht, wie ich dann errechnete Werte in den `data:`-Block unterbringen kann. Wo finde ich ein Beispiel für "... mit den finalen Werten aufrufen..."?
Nebenbei: Auch wenn meine Fragerei nörgelnd und kritisch wirkt, ich finde die Integration sehr gut, sonst würde ich mich ja nicht darum so sehr bemühen.

**Reply by marq24** [reply]
...das ist aber doch genau mein Punkt (bzgl. Deines Links): Du musst die Templates in Automations verwenden, die dann eine Aktion aufrufen... (und dabei die 'Templates zu Werten' von der Automation zu den Werten aufgelöst und dann als Werte an die Aktion übergeben) - Du kannst aber eben nicht direkt Template Code beim manuellen start eine Aktion verwenden... 

Unter 'Entwicklerwerkzeugen/Aktionen' kannst Du (m.E. nach) keine Templates verwenden - sonder nur unter `Einstellungen/Automationen`
